packages/mdc-ripple/foundation.js

```diff
@@ -488,17 +485,20 @@ class MDCRippleFoundation extends MDCFoundation {
   /** @private */
   layoutInternal_() {
     this.frame_ = this.adapter_.computeBoundingRect();
-
     const maxDim = Math.max(this.frame_.height, this.frame_.width);
-    const surfaceDiameter = Math.sqrt(Math.pow(this.frame_.width, 2) + Math.pow(this.frame_.height, 2));
 
-    // 60% of the largest dimension of the surface
-    this.initialSize_ = maxDim * MDCRippleFoundation.numbers.INITIAL_ORIGIN_SCALE;
+    // Surface diameter is treated differently for unbounded vs. bounded ripples.
+    // Unbounded ripple diameter is calculated smaller since the surface is expected to already be padded appropriately
+    // to extend the pointer target, and the ripple is expected to meet the edges of the padded hitbox
+    // (which is typically square). Bounded ripples, on the other hand, are fully expected to expand beyond the
+    // surface's longest diameter (calculated based on the diagonal plus a constant padding).
+    this.maxRadius_ = this.adapter_.isUnbounded() ? maxDim :
+      Math.sqrt(Math.pow(this.frame_.width, 2) + Math.pow(this.frame_.height, 2)) + MDCRippleFoundation.numbers.PADDING;
```

Comment:
It might be more readable to refactor the "bounded" expression a bit:

```js
const getDiagonal = () => {
  const hypotenuse = Math.sqrt(Math.pow(this.frame_.width, 2) + Math.pow(this.frame_.height, 2));
  return hypotenuse + MDCRippleFoundation.numbers.PADDING;
};

this.maxRadius_ = this.adapter_.isUnbounded() ? maxDim : getDiagonal();
```

Or `getDiagonal` could be a separate private method if you prefer.

Reply:
I'm normally against this sort of thing for a single use, but...maths, so yeah.